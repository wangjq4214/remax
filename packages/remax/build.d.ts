export { build as buildApp } from '@remax/cli/lib/build';
