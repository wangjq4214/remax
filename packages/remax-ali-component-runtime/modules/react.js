import { r as react } from './index-c3dc9930.js';

var Children = react.Children;
var Component = react.Component;
var Fragment = react.Fragment;
var Profiler = react.Profiler;
var PureComponent = react.PureComponent;
var StrictMode = react.StrictMode;
var Suspense = react.Suspense;
var cloneElement = react.cloneElement;
var createContext = react.createContext;
var createElement = react.createElement;
var createFactory = react.createFactory;
var createRef = react.createRef;
var forwardRef = react.forwardRef;
var isValidElement = react.isValidElement;
var lazy = react.lazy;
var memo = react.memo;
var unstable_SuspenseList = react.unstable_SuspenseList;
var unstable_withSuspenseConfig = react.unstable_withSuspenseConfig;
var useCallback = react.useCallback;
var useContext = react.useContext;
var useDebugValue = react.useDebugValue;
var useEffect = react.useEffect;
var useImperativeHandle = react.useImperativeHandle;
var useLayoutEffect = react.useLayoutEffect;
var useMemo = react.useMemo;
var useReducer = react.useReducer;
var useRef = react.useRef;
var useState = react.useState;
var version = react.version;
export {
  Children,
  Component,
  Fragment,
  Profiler,
  PureComponent,
  StrictMode,
  Suspense,
  cloneElement,
  createContext,
  createElement,
  createFactory,
  createRef,
  forwardRef,
  isValidElement,
  lazy,
  memo,
  unstable_SuspenseList,
  unstable_withSuspenseConfig,
  useCallback,
  useContext,
  useDebugValue,
  useEffect,
  useImperativeHandle,
  useLayoutEffect,
  useMemo,
  useReducer,
  useRef,
  useState,
  version,
};
