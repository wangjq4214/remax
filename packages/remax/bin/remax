#!/usr/bin/env node
require('@remax/cli').run(process.argv.slice(2, process.argv.length));
