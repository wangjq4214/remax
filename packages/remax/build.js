module.exports.buildApp = require('@remax/cli/lib/build').build;
