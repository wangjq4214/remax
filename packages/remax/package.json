{
  "name": "remax",
  "version": "2.7.3",
  "description": "Remax 是一个全新的基于 React 的小程序开发框架",
  "main": "index.js",
  "bin": {
    "remax": "./bin/remax"
  },
  "keywords": [
    "react",
    "miniapp",
    "mini-program",
    "wechat"
  ],
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/remaxjs/remax/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/remaxjs/remax.git"
  },
  "homepage": "https://remaxjs.org",
  "dependencies": {
    "@remax/ali": "2.7.3",
    "@remax/cli": "2.7.3",
    "@remax/macro": "2.7.3",
    "@remax/one": "2.7.3",
    "@remax/runtime": "2.7.3",
    "@remax/toutiao": "2.7.3",
    "@remax/web": "2.7.3",
    "@remax/wechat": "2.7.3",
    "mini-types": "^0.1.1",
    "miniprogram-api-typings": "^2.10.4"
  },
  "gitHead": "06dcaa3df31f6bde22b6d07a979c361dc137351f"
}
