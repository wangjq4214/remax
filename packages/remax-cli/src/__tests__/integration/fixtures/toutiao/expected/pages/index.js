require('./../runtime.js');
(tt["webpackJsonp"] = tt["webpackJsonp"] || []).push([[2],[
/* 0 */,
/* 1 */,
/* 2 */
/***/ (function(module, exports) {

module.exports = require("@remax/runtime");

/***/ }),
/* 3 */,
/* 4 */,
/* 5 */,
/* 6 */
/***/ (function(module, exports) {

module.exports = require("react");

/***/ }),
/* 7 */
/***/ (function(module, exports, __webpack_require__) {

module.exports = __webpack_require__(8);


/***/ }),
/* 8 */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _remax_runtime__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(2);
/* harmony import */ var _remax_runtime__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(_remax_runtime__WEBPACK_IMPORTED_MODULE_0__);
/* harmony import */ var regenerator_runtime__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(9);
/* harmony import */ var regenerator_runtime__WEBPACK_IMPORTED_MODULE_1___default = /*#__PURE__*/__webpack_require__.n(regenerator_runtime__WEBPACK_IMPORTED_MODULE_1__);
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(6);
/* harmony import */ var react__WEBPACK_IMPORTED_MODULE_2___default = /*#__PURE__*/__webpack_require__.n(react__WEBPACK_IMPORTED_MODULE_2__);
/* harmony import */ var remax_toutiao__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(10);
/* harmony import */ var remax_toutiao__WEBPACK_IMPORTED_MODULE_3___default = /*#__PURE__*/__webpack_require__.n(remax_toutiao__WEBPACK_IMPORTED_MODULE_3__);
/* harmony import */ var remax_wechat__WEBPACK_IMPORTED_MODULE_4__ = __webpack_require__(11);
/* harmony import */ var remax_wechat__WEBPACK_IMPORTED_MODULE_4___default = /*#__PURE__*/__webpack_require__.n(remax_wechat__WEBPACK_IMPORTED_MODULE_4__);



function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }





var _page = function _page() {
  var props = {};
  var TextElement = /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__["cloneElement"]( /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__["createElement"](remax_toutiao__WEBPACK_IMPORTED_MODULE_3__["Text"], null));

  function handleClick() {
    return _handleClick.apply(this, arguments);
  }

  function _handleClick() {
    _handleClick = _asyncToGenerator( /*#__PURE__*/regenerator_runtime__WEBPACK_IMPORTED_MODULE_1___default.a.mark(function _callee() {
      return regenerator_runtime__WEBPACK_IMPORTED_MODULE_1___default.a.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return Promise.resolve();

            case 2:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return _handleClick.apply(this, arguments);
  }

  function handleTouchStart() {}

  return /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__["createElement"](remax_toutiao__WEBPACK_IMPORTED_MODULE_3__["View"], null, /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__["createElement"](remax_toutiao__WEBPACK_IMPORTED_MODULE_3__["View"], _extends({
    onClick: handleClick,
    onTouchStart: handleTouchStart,
    id: "view",
    "data-foo": "bar"
  }, props), "foo"), TextElement, /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__["createElement"](remax_wechat__WEBPACK_IMPORTED_MODULE_4__["FunctionalPageNavigator"], null), /*#__PURE__*/react__WEBPACK_IMPORTED_MODULE_2__["createElement"](UnBindingComponent, null));
};

/* harmony default export */ __webpack_exports__["default"] = (Page(Object(_remax_runtime__WEBPACK_IMPORTED_MODULE_0__["createPageConfig"])(_page, "pages/index")));

/***/ }),
/* 9 */
/***/ (function(module, exports) {

module.exports = require("regenerator-runtime");

/***/ }),
/* 10 */
/***/ (function(module, exports) {

module.exports = require("remax/toutiao");

/***/ }),
/* 11 */
/***/ (function(module, exports) {

module.exports = require("remax/wechat");

/***/ })
],[[7,0]]]);