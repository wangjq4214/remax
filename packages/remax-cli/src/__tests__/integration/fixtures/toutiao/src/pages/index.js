import * as React from 'react';
import { View, Text } from 'remax/toutiao';
import { FunctionalPageNavigator as UnknownComponent } from 'remax/wechat';

export default () => {
  const props = {};
  const TextElement = React.cloneElement(<Text />);

  async function handleClick() {
    await Promise.resolve();
  }

  function handleTouchStart() {}

  return (
    <View>
      <View onClick={handleClick} onTouchStart={handleTouchStart} id="view" data-foo="bar" {...props}>
        foo
      </View>
      {TextElement}
      <UnknownComponent />
      <UnBindingComponent />
    </View>
  );
};
