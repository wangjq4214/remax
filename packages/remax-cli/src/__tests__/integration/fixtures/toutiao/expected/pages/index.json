{
  "navigationBarBackgroundColor": "#323239",
  "navigationBarTitleText": "Toutiao App",
  "usingComponents": {}
}